### PR TITLE
[15.0][ADD] stock_picking_info_lot: lot information additional field on Stock Transfers

### PR DESCRIPTION
**Pull request by ZavarinArtem**
New stock_picking_info_lot module

### CODE REVIEW

**Comment by dreispt** [comment]
Thank you @ZavarinArtem 

Doing functional tests I  found an issue:
Validating a PO receipt works even if the products are "lot info required" but don't have any value set on the lot info field.

Thanks

File: stock_picking_info_lot/readme/DESCRIPTION.rst [hunk 1]
```diff
@@ -0,0 +1,4 @@
+This feature allows for inventory operations to have additional information about the lot numbers being moved.
```
Comment (rousseldenis) [inline stock_picking_info_lot/readme/DESCRIPTION.rst]: @ZavarinArtem  Could you a little bit be more accurate about 'additional information' is ?

Comment (ZavarinArtem) [reply stock_picking_info_lot/readme/DESCRIPTION.rst]: Thank you for the review.
I've rewritten the description, can you look is it enough information now?

File: stock_picking_info_lot/models/stock_move_line.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
+from odoo import fields, models
+
+
+class StockMoveLine(models.Model):
+    _inherit = "stock.move.line"
+
+    lot_info = fields.Char("Lot/Serial Number info")
```
Comment (dreispt) [inline stock_picking_info_lot/models/stock_move_line.py]:
```suggestion
    lot_info = fields.Char("Lot/Serial Number Info")
```

File: stock_picking_info_lot/models/product_template.py [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+
+class ProductTemplate(models.Model):
+
+    _inherit = "product.template"
+
+    lot_info_usage = fields.Selection(
+        [("no", "No"), ("optional", "Optional"), ("required", "Required")],
+        default="no",
+        string="Lot information usage",
```
Comment (dreispt) [inline stock_picking_info_lot/models/product_template.py]:
```suggestion
        string="Lot Information Usage",
```

File: stock_picking_info_lot/models/stock_move_line.py [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+    lot_info_usage = fields.Selection(
+        [("no", "No"), ("optional", "Optional"), ("required", "Required")],
+        related="product_id.product_tmpl_id.lot_info_usage",
+    )
```
Comment (dreispt) [inline stock_picking_info_lot/models/stock_move_line.py]: Fix the log warning `odoo.fields: stock.move.line.lot_info_usage: selection attribute will be ignored as the field is related`
```suggestion
    lot_info_usage = fields.Selection(related="product_id.product_tmpl_id.lot_info_usage")
```

**Comment by dreispt** [comment]
@ZavarinArtem see https://github.com/profispoc/stock-logistics-workflow/pull/6

**Comment by ZavarinArtem** [comment]
> @ZavarinArtem see [profispoc#6](https://github.com/profispoc/stock-logistics-workflow/pull/6)

@dreispt Done. Sorry you had to do it for me.

**Comment by dreispt** [comment]
@ZavarinArtem No problem.
We now have some logic worth a Unit Test 😃 
Would you like to have a try at that?

**Comment by ZavarinArtem** [comment]
@dreispt Added test for validation, as much as I understand how tests should work. Could you review it?

**Review by dreispt — APPROVED** [review]
Looks great!

**Comment by dreispt** [comment]
Looks great. Now it is best to squash all the work commits into a single one using `git rebase -i 15.0`. Thanks!

**Review by AnilKesariya — APPROVED** [review]

File: stock_picking_info_lot/readme/DESCRIPTION.rst [hunk 5]
```diff
@@ -0,0 +1,4 @@
+This feature adds a text field that allows for inventory operations (i.e. picking out) to have additional information about the lot numbers being moved.
```
Comment (rousseldenis) [inline stock_picking_info_lot/readme/DESCRIPTION.rst:1]: @ZavarinArtem @dreispt Shouldn't this behaviour be activated through an option on picking type level? As at some flow points, you don't have the information nor want to fill in at that time.

Comment (dreispt) [reply stock_picking_info_lot/readme/DESCRIPTION.rst:1]: @rousseldenis Can you also do that with Odoo's Lot/Serial numbers?

Comment (rousseldenis) [reply stock_picking_info_lot/readme/DESCRIPTION.rst:1]: Of course :

image

Comment (dreispt) [reply stock_picking_info_lot/readme/DESCRIPTION.rst:1]: Thanks, I agree that is a relevant feature.
It does mean non trivial extra development, so I don't think we can factor that it.
But it makes sense to add that to the module's roadmap.

**Comment by ZavarinArtem** [comment]
@dreispt I think it should be fine now.

File: stock_picking_info_lot/readme/DESCRIPTION.rst (hunk 5 above)
Comment (rousseldenis) [reply stock_picking_info_lot/readme/DESCRIPTION.rst:1]: @ZavarinArtem Please add this in ROADMAP.rst

Comment (ZavarinArtem) [reply stock_picking_info_lot/readme/DESCRIPTION.rst:1]: @rousseldenis Done

**Review by rousseldenis — APPROVED** [review]

**Pull request merged**